jujugui/static/gui/src/app/store/charmstore-api.js

```diff
@@ -284,16 +284,20 @@ YUI.add('charmstore-api', function(Y) {
         successfully.
       @param {Function} failureCallback Called when the api request fails
         with a response of >= 400.
+      @param {Integer} limit The number of results to get.
     */
-    search: function(filters, successCallback, failureCallback) {
-      var defaultFilters =
-                        '&limit=30&' +
```

Comment:
Instead of the conditional below you could instead just do

`'&limit=' + limit || 30 + '&' +`